lib/sycamore/sycamore/writers/opensearch.py

```diff
@@ -82,65 +189,21 @@ def extract_os_document(data):
             "bbox": None,
             "shingles": None,
         }
+        data = document.data
         for k, v in default.items():
             if k in data:
                 result[k] = data[k]
             else:
                 result[k] = v
-        return result
-
-    @timetrace("OsrchWrite")
```

Comment:
recover the timetrace; Might need to use the non-decorator use since in the writer base class, you want to timetrace with `self.__class__.__name__`